code/datums/components/riding/riding.dm

```diff
@@ -207,6 +208,10 @@
 		return !forbid_turf_typecache[current.type]
 	return TRUE
 
+/// Every time the driver tries to move, this is called to see if they can actually drive and move the vehicle (via relaymove)
+/datum/component/riding/proc/driver_move(atom/movable/movable_parent, mob/living/user, direction)
```

Comment:
Why don't you have the subtypes that actually use this register the signal themselves?